src/components/Reports/FourteenMonthReports/SalaryReport/SalaryReportTable.tsx

```diff
@@ -0,0 +1,330 @@
+  b: { [key in Key]: number | string },
+) => number {
+  return order === 'desc'
+    ? (a, b) => descendingComparator(a, b, orderBy)
+    : (a, b) => -descendingComparator(a, b, orderBy);
+}
+
+function stableSort<T>(array: T[], comparator: (a: T, b: T) => number) {
+  const stabilizedThis = array.map((el, index) => [el, index] as [T, number]);
+  stabilizedThis.sort((a, b) => {
+    const order = comparator(a[0], b[0]);
+    if (order !== 0) return order;
+    return a[1] - b[1];
+  });
+  return stabilizedThis.map((el) => el[0]);
+}
+
+export const SalaryReportTable: React.FC<Props> = ({
+  accountListId,
+  isNavListOpen,
+  title,
+  onNavListToggle,
+  ...rest
+}) => {
+  const [isExpanded, setExpanded] = useState<boolean>(false);
+  const [order, setOrder] = useState<Order>('asc');
+  const [orderBy, setOrderBy] = useState<string | number | null>(null);
+  const reportTableRef = useRef(null);
+  const { t } = useTranslation();
+
+  const isMobile = useMediaQuery((theme: Theme) =>
+    theme.breakpoints.down('sm'),
+  );
+
+  const { data, loading, error } = useFourteenMonthReportQuery({
+    variables: {
+      accountListId,
+    },
+  });
+
+  const handlePrint = useReactToPrint({
+    content: () => reportTableRef.current,
+  });
+
+  const contacts = useMemo(() => {
+    return data?.fourteenMonthReport.currencyGroups?.flatMap(
+      (currencyGroup) => [...currencyGroup?.contacts],
+    );
+  }, [data?.fourteenMonthReport.currencyGroups]);
+
+  const handleExpandToggle = (): void => {
+    setExpanded((prevExpanded) => !prevExpanded);
+  };
+
+  const handleRequestSort = (
+    event: React.MouseEvent<unknown>,
+    property: string | number,
+  ) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? 'desc' : 'asc');
+    setOrderBy(property);
+  };
+
+  return (
+    <Box>
+      <StickyHeader p={2}>
+        <Grid
+          container
+          justify={isMobile ? 'center' : 'space-between'}
+          spacing={2}
+          {...rest}
+        >
+          <Grid item>
+            <Box display="flex" alignItems="center">
+              <NavListButton
+                panelopen={isNavListOpen ? 1 : 0}
+                onClick={onNavListToggle}
+              >
+                <NavListIcon titleAccess={t('Toggle Filter Panel')} />
+              </NavListButton>
+              <HeaderTitle variant="h5">{t(title)}</HeaderTitle>
+            </Box>
+          </Grid>
+          <Grid item>
+            <ButtonGroup aria-label="report header button group">
+              <Button
+                startIcon={
+                  <SvgIcon fontSize="small">
+                    <CodeIcon />
+                  </SvgIcon>
+                }
+                onClick={handleExpandToggle}
+              >
+                {t(
+                  `${isExpanded ? 'Hide' : 'Expand'}${
+                    isMobile ? '' : ' Partner Info'
+                  }`,
+                )}
```

Comment:
Translation keys can't be dynamic. You can just flip where you're calling `t`.
```suggestion
                {isExpanded ? t('Hide') : t('Expand')}
                {isMobile ? '' : t('Partner Info')}
```